src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java

```diff
@@ -306,7 +306,9 @@ private static String createPropertiesText() {
 
             String typeText = "java.lang.String[]";
             final String propertyType = property.get(2).getTextContent();
-            final boolean isSpecialAllTokensType = propertyType.contains("set of any supported");
+            // until we migrate descendanttoken.xml xdoc to use macro properties
+            final boolean isSpecialAllTokensType = propertyType.contains("set of any supported")
+                    || "TokenTypes".equals(propertyType);
```

Comment:
To avoid forgotten until, we put link to issue, please create separate issue for my migration of this module to to macros,  and reference it this comment instead of text.

Reply:
> To avoid forgotten until, we put link to issue, please create separate issue for my migration of this module to to macros, and reference it this comment instead of text.

done